.gitlab/binary_build/system_probe.yml

```diff
@@ -12,9 +12,7 @@
     - find "$CI_BUILDS_DIR" ! -path '*DataDog/datadog-agent*' -delete || true  # Allow failure, we can't remove parent folders of datadog-agent
   script:
     - inv check-go-version
-    - inv -e system-probe.build --strip-object-files
-    # fail if references to glibc >= 2.18
-    - objdump -p $CI_PROJECT_DIR/$SYSTEM_PROBE_BINARIES_DIR/system-probe | egrep 'GLIBC_2\.(1[8-9]|[2-9][0-9])' && exit 1
```

Comment:
What happened to this? We need this safeguard to ensure we don't accidentally start using a newer glibc.

Reply:
Basically this has never worked, since we have a safeguard on the wrong binary. Happy to spawn another PR to add this safeguard on the actually binary shipped, the one built by omnibus

Reply:
The omnibus building of the system-probe binary is a somewhat recent addition. They could mismatch, from a glibc standpoint, if the buildimages differed.

Reply:
I will spawn a PR to add this check you are right, better be safe on this front